runtime/j9vm/jvm.c

```diff
@@ -859,7 +859,7 @@ getj9bin()
 	/* misnamed - returns the directory that the jvm DLL is found in, NOT the directory that the J9 VM itself is in. */
 
 	J9StringBuffer *result = NULL;
-#if defined(LINUX) || defined(OSX) && !defined(J9ZTPF)
+#if (defined(LINUX) || defined(OSX)) && !defined(J9ZTPF)
```

Comment:
Can you clarify how this is a `fix`?  A || B && !C is the same result with or without the bracket.

Further, ZTPF logically connects to LINUX, not to OSX.  If you were going to change this, it would be 
`#if ((defined(LINUX) && !defined(J9ZTPF)) || defined(OSX)`

Reply:
`&&` has higher precedence than `||` so the old condition was equivalent to
```
#if defined(LINUX) || (defined(OSX) && !defined(J9ZTPF))
```
@DanHeidinga If ZTPF is more closely related to LINUX, then your proposal would be better.

Reply:
Thanks for clarifying the precedence rules.  That's great info to include when creating this kind of PR in the future.  Thanks @keithc-ca